frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget.jsx

```diff
@@ -10,34 +10,47 @@ import GroupSelect from "metabase/admin/people/components/GroupSelect";
 import LoadingSpinner from "metabase/components/LoadingSpinner";
 import Modal from "metabase/components/Modal";
 import { PermissionsApi, SettingsApi } from "metabase/services";
-import { isDefaultGroup } from "metabase/lib/groups";
+import { isDefaultGroup, isAdminGroup } from "metabase/lib/groups";
 
 import SettingToggle from "./SettingToggle";
+import DeleteGroupMappingModal from "./GroupMappingsWidget/DeleteGroupMappingModal";
 
 const groupIsMappable = group => !isDefaultGroup(group);
 
 export default class GroupMappingsWidget extends React.Component {
   constructor(props, context) {
     super(props, context);
     this.state = {
+      showDeleteMappingModal: false,
       showEditModal: false,
       showAddRow: false,
       groups: null,
       mappings: {},
+      savedMappings: {},
       saveError: null,
+      dnForVisibleDeleteMappingModal: null,
+      groupIdsForVisibleDeleteMappingModal: null,
+      whenDeletingMappingGroups: {
+        groupsToClearAllPermissions: [],
+        groupsToDelete: [],
+      },
```

Comment:
Doesn't seem like it's used while rendering. Should it live inside the component state?

Reply:
Good call, thanks.

I'll check on answers to https://github.com/metabase/metabase/pull/27664#discussion_r1084698148 so I can tweak this, maybe we will inline the widget in this PR, or in a follow-up PR.